libs/utils/android/workloads/geekbench.py

```diff
@@ -0,0 +1,155 @@
+
+        :param out_dir: Path on host to experiment directory where
+                        to store results.
+        :type out_dir: str
+
+        :param test_name: Name of the test to run
+        :type test_name: str
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+            - 'systrace'
+            - 'ftrace'
+            - any combination of the above in a space-separated string.
+        :type collect: list(str)
+        """
+
+        # Initialize energy meter results
+        self.out_dir = out_dir
+        self.collect = collect
+
+        # Regexps for benchmark synchronization
+        start_logline = r'ActivityManager: Start.*'\
+                         'com.primatelabs.geekbench'
+        GEEKBENCH_BENCHMARK_START_RE = re.compile(start_logline)
+        self._log.debug("START string [%s]", start_logline)
+
+        end_logline = r'GEEKBENCH_RESULT: (.*)'
+        GEEKBENCH_BENCHMARK_END_RE = re.compile(end_logline)
+        self._log.debug("END string [%s]", end_logline)
+
+        # Clear the stored data for the application, so we always start with
+        # an EULA to clear
+        self._target.execute('pm clear {}'.format(self.package))
+
+        # Clear logcat from any previous runs
+        # do this on the target as then we don't need to build a string
+        self._target.clear_logcat()
+
+        # Unlock device screen (assume no password required)
+        System.menu(self._target)
+        # Press Back button to be sure we run the benchmark from the start
+        System.back(self._target)
+
+        # Force screen in PORTRAIT mode
+        Screen.set_orientation(self._target, portrait=True)
+
+        # Set min brightness
+        Screen.set_brightness(self._target, auto=False, percent=0)
+
+        # Start app on the target device
+        System.start_activity(self._target, self.package, self.activity)
+        # Allow the activity to start
+        sleep(2)
+
+        # Parse logcat output lines to find beginning and end
+        logcat_cmd = self._adb(
+                'logcat ActivityManager:* System.out:I *:S GEEKBENCH_RESULT:*'\
+                .format(self._target.adb_name))
+        self._log.info("%s", logcat_cmd)
+        logcat = Popen(logcat_cmd, shell=True, stdout=PIPE)
+
+        # Click to accept the EULA
+        System.tap(self._target, 73, 55)
+        sleep(1)
+
+        # The main window opened will have the CPU benchmark
+        # Swipe to get the COMPUTE one
+        if test_name.upper() == 'COMPUTE':
+            System.hswipe(self._target, 10, 80, duration=100, swipe_right=False)
+
+        # Press the 'RUN <test_name> BENCHMARK' button
+        System.tap(self._target, 73, 72)
+
+        while True:
+
+            # read next logcat line (up to max 1024 chars)
+            message = logcat.stdout.readline(1024)
+
+            # Benchmark start trigger
+            if GEEKBENCH_BENCHMARK_START_RE.search(message):
```

Comment:
nit-pick: I would prefer a sequence of:

```python
match = GEEKBENCH_BENCHMARK_START_RE.search(message)
if match:
    ....
match = GEEKBENCH_BENCHMARK_END_RE.search(message)
if match:
   ....
```

Which makes the code slightly less optimized but that's not running on target and makes also the code more simple to read. That's a pattern we want users to easily copy-adapt to their workloads. In case they should need to match multiple REs (like we do with for jankbenck) we can easily end up with many horrible indentation levels.